### PR TITLE
[GH1408] Update time fields correctly in sync operations

### PR DESCRIPTION
**Pull request by ronys**
Fix for #1408 is currently Windows-only. Needs to be applied to wx as well.

### CODE REVIEW

File: src/core/Item.h [hunk 1]
```diff
@@ -124,6 +124,13 @@ class CItem
     LAST_FIELD
   };
 
+  static bool IsTimeField(int ft)
```
Comment (rafaelx) [inline src/core/Item.h:127]: Although the _FieldType_ data type exists as an enumeration, the API uses an integer type instead. Is this for historical reasons, programming reasons, or is a C API intended to be provided to make it easier for other projects to create bindings?

Comment (ronys) [reply src/core/Item.h:127]: Historical reasons. Today I'd keep this as a reminder that the values passed may be outside those defined in the enum, as in forward compatibility. That is, reading a database written on a newer version that has fields (enums) that are unrecognized by the reading version. PasswordSafe will read and write these fields, but won't know how to display them.

File: src/ui/Windows/MainFile.cpp [hunk 2]
```diff
@@ -3778,10 +3778,16 @@ LRESULT DboxMain::SynchCompareResult(PWScore *pfromcore, PWScore *ptocore,
              sxSync_DateTime, IDSC_SYNCPOLICY);
         if (pPolicyCmd != NULL)
           pmulticmds->Add(pPolicyCmd);
+      } else if (i == 8) { // XXX hack test theory
```
Comment (dskrvk) [inline src/ui/Windows/MainFile.cpp:3781]: Did you mean to keep this XXX comment? If so, it should have more context to help future maintainers.

Also, is 8 one of the values of `FieldType` - could we use that instead?

Comment (ronys) [reply src/ui/Windows/MainFile.cpp:3781]: Oops - wrong commit pushed...
More work due in any case - thanks!

**Review by dskrvk — APPROVED** [review]

**Pull request merged**